pkg/stream/retry_io.go

```diff
@@ -0,0 +1,100 @@
+package stream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/cloud/aws/sqs"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+type RetryHandlerIO struct {
+	input    AcknowledgeableInput
+	output   Output
+	settings *RetrySettings
+}
+
+func NewRetryHandlerIO(ctx context.Context, config cfg.Config, logger log.Logger, settings *RetrySettings) (RetryHandler, error) {
```

Comment:
I would make the type private and return an interface if possible

Reply:
it is returning an interface already? :thinking:

Reply:
Seems like I can't read, you are right